src/Rule/Validate/Ip.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ *
+ * This file is part of the Aura project for PHP.
+ *
+ * @package Aura.Filter
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+namespace Aura\Filter\Rule\Validate;
+
+/**
+ *
+ * Validates a value is an IP address.
+ *
+ * @package Aura.Filter
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+class Ip
+{
+    /**
+     *
+     * Validates that the value is a legal IP address.
+     *
+     * @return bool True if valid, false if not.
+     *
+     */
+    public function __invoke($subject, $field)
+    {
+        $value = $subject->$field;
+
+        // This validates without regard to reserved or private ranges in both v4 and v6
+        if (filter_var($value, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6) === false) {
+            return false;
```

Comment:
Thank you @auroraeosrose for clarifying.

@BallisticPain all is good .

Thank you.